test/TestINode.java

```diff
@@ -0,0 +1,38 @@
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class TestINode {
+	
+	WordsEnvironment environment = new WordsEnvironment();
+
+	/*********************
+	 * EvalPosition 
+	 ********************/
+	AST numLeaf1 = new LNode(AST.ASTType.NUM, 0, 0.0);
+	AST numLeaf2 = new LNode(AST.ASTType.NUM, 0, 2.1);
+	AST stringLeafCoercable = new LNode(AST.ASTType.STRING, 0, "2.1");
```

Comment:
JUnit has a fun feature where all instance variables are actually reinitialized between each test case. I've modified the TestEnvironment file to better document and show this. 

Though you have a fair point that it's not really necessary to test coercion separately in each INode test case, as long as we have extensive testing of coercion. I'm going to write up these test cases real fast.